src/core/ext/filters/client_channel/parse_address.cc

```diff
@@ -56,7 +58,20 @@ bool grpc_parse_unix(const grpc_uri* uri,
   if (path_len == maxlen) return false;
   un->sun_family = AF_UNIX;
   strcpy(un->sun_path, uri->path);
+#ifdef GRPC_HAVE_ABSTRACT_UNIX_SOCKET
```

Comment:
I would avoid that pattern of `#ifdef` completely. This codepath should be no adverse effect on platforms that do not support abstract unix domain sockets, aside from the fact it simply won't work if passing something with an '@' in the name, which wouldn't change anything from the other codepath.

Reply:
I actually specifically added them for the PR :)

I thought that the idea would be: on Linux it becomes an abstract unix socket, however, on all other unix systems it is still just a regular socket file.

That said, if the consensus is to remove that distinction, I'm more than happy to remove this again. IMHO `#ifdef`s don't help with readability of the code, and it makes things a lot easier without them of course.

Reply:
```suggestion
#ifdef GRPC_HAVE_ABSTRACT_UNIX_SOCKET
```

Reply:
Would this be preferable, to unify the code paths?
```suggestion
if (un->sun_path[0] == '@') {
  if (!GRPC_HAVE_ABSTRACT_UNIX_SOCKET) {
    gpr_log(GPR_ERROR, "Abstract Unix Sockets not supported on this platform");
    return false;
  }
  un->sun_path[0] = '\0';
}
resolved_addr->len = static_cast<socklen_t>(sizeof(un->sun_family) + path_len);
```

Reply:
I think what Nico is suggesting is to just remove the `#ifdef` and otherwise leave the code as-is.  If the system does not support abstract unix sockets, then the call will not work, but we don't have to figure that out at compile time.